### PR TITLE
DOC-9335 -- P2P Xamarin upgrade to 3.0 Add workaround for using emula…

### PR DESCRIPTION
**Pull request by ibsoln**
…tors

https://issues.couchbase.com/browse/DOC-9335

DOC-9335 -- Upgrade to 3.0

DOC-9335 -- Upgrade to 3.0

DOC-9335 - Upgrade to 3.0 fix module variable

DOC-9335 - Upgrade to 3.0

post-revert

DOC-9335 - updated samples

DOC-9335 -- P2P Xamarint upgrade to 3.0
https://issues.couchbase.com/browse/DOC-9335

DOC-9335 -- P2P Xamarin upgrade to 3.0
Add workaround for using emulators
https://issues.couchbase.com/browse/DOC-9335

DOC-9335 -- P2P Xamarin upgrade to 3.0
Add workaround for using emulators
https://issues.couchbase.com/browse/DOC-9335

DOC-9335 -- P2P Xamarin upgrade to 3.0
Add workaround for using emulators
https://issues.couchbase.com/browse/DOC-9335

### CODE REVIEW

**Review by rajagp — APPROVED** [review]

File: content/modules/cbl-p2p-sync-websockets/pages/dotnet/cbl-p2p-sync-websockets.adoc [hunk 1]
```diff
@@ -299,99 +449,180 @@ include::example$code-samples.cs[tags=ListenerValidationTestMode, indent=0]
...
+=== Connecting to an Android emulator
 
-=== Connecting to Android emulator over localhost
-When starting a listener on Android emulator and trying to connect it from another emulator or iOS simulator on localhost, the following steps must be followed
+IMPORTANT: Sync will not work between two emulators. At least one app must be running on a device.
 
-. Setup port forwarding. +
+When starting a listener on Android emulator and trying to connect it from a device or iOS simulator on localhost, the following steps must be followed:
+
+. Hard-code a port so you know what EP to use
```
Comment (rajagp) [inline content/modules/cbl-p2p-sync-websockets/pages/dotnet/cbl-p2p-sync-websockets.adoc:598]: @ibsoln  Is "EP" supposed to be endpoint. If so, pls. expand. Its not an acronym.  Applies globally

File: content/modules/cbl-p2p-sync-websockets/pages/dotnet/cbl-p2p-sync-websockets.adoc [hunk 2]
```diff
@@ -299,99 +449,180 @@ include::example$code-samples.cs[tags=ListenerValidationTestMode, indent=0]
...
+. Start the App on your device or iOS simulator
```
Comment (rajagp) [inline content/modules/cbl-p2p-sync-websockets/pages/dotnet/cbl-p2p-sync-websockets.adoc:599]: @ibsoln  Shouldn't this be Android emulator and not iOS simulator

**Review by rajagp — CHANGES_REQUESTED** [review]

File: content/modules/cbl-p2p-sync-websockets/pages/dotnet/cbl-p2p-sync-websockets.adoc (hunk 1 above)
Comment (ibsoln) [reply content/modules/cbl-p2p-sync-websockets/pages/dotnet/cbl-p2p-sync-websockets.adoc:598]: > 

It is. I'll fix it.

File: content/modules/cbl-p2p-sync-websockets/pages/dotnet/cbl-p2p-sync-websockets.adoc (hunk 2 above)
Comment (ibsoln) [reply content/modules/cbl-p2p-sync-websockets/pages/dotnet/cbl-p2p-sync-websockets.adoc:599]: Two app instances are needed. As we can't have two emulators, I've assumed one emulator and either a device (iOS or Android) and one emulator. Here we are starting an iOS simulator on which we will later browse for a listener.

**Pull request merged**